### PR TITLE
chore(test): increase test coverage for async-rewriter

### PR DESCRIPTION
**Pull request by rose-m**
This covers edge cases and removes essentially duplicate code to increase coverage for `async-rewriter`.

image

### CODE REVIEW

**Review by addaleax — APPROVED** [review]
LGTM, thank you! :slightly_smiling_face:

File: packages/async-rewriter/src/symbol-table.ts [hunk 1]
```diff
@@ -93,21 +93,15 @@ export default class SymbolTable {
    * Do not deep copy path attributes as they must stay as references to the AST.
    */
   public deepCopy(): SymbolTable {
-    const newStack = [];
```
Comment (aherlihy) [inline packages/async-rewriter/src/symbol-table.ts:96]: Could you explain why this was removed?

Comment (aherlihy) [reply packages/async-rewriter/src/symbol-table.ts:96]: Oh I see, sorry I misread the diff, it's just repeated code

**Review by aherlihy — APPROVED** [review]
Nice!

**Comment by addaleax** [comment]
> ```
> File                                   | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
> ---------------------------------------|---------|----------|---------|---------|-------------------
> All files                              |   96.02 |    85.62 |   95.05 |   96.09 |

… nice! :)
